File: CS/ChartsDemo.Wpf/ViewModels/DailyWeatherViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Globalization;
using System.Windows.Media;
using System.Xml.Linq;
using DevExpress.Mvvm;
using DevExpress.Xpf.Charts;

namespace ChartsDemo {
    public class DailyWeatherViewModel {
        const string vostokStationName = "Vostok Station";
        const string deathValleyName = "Death Valley, NV";

        readonly List<WeatherItem> weather;

        public List<WeatherItem> Weather { get { return weather; } }

        public DailyWeatherViewModel() {
            List<WeatherRecord> valleyData = LoadWeatherData("DeathValley.xml");
            List<WeatherRecord> vostokData = LoadWeatherData("VostokStation.xml");

            Palette palette = new OfficePalette();
            weather = new List<WeatherItem>() {
                new WeatherItem(valleyData, false, palette[1], deathValleyName),
                new WeatherItem(valleyData, true, palette[1], deathValleyName),
                new WeatherItem(vostokData, false, palette[0], vostokStationName),
                new WeatherItem(vostokData, true, palette[0], vostokStationName),
            };
        }
        List<WeatherRecord> LoadWeatherData(string fileName) {
            List<WeatherRecord> items = new List<WeatherRecord>();
            XDocument weatherDocument = DataLoader.LoadXmlFromResources("/Data/DailyWeather/" + fileName);
            foreach (XElement element in weatherDocument.Root.Elements("Weather")) {
                items.Add(WeatherRecord.Load(element));
            }
            return items;
        }
        public virtual IChartAnimationService ChartAnimationService { get { return null; } }
        public void OnModuleLoaded() {
            if (ChartAnimationService != null)
                ChartAnimationService.Animate();
        }
    }

    public class WeatherItem : BindableBase {
        public int AverageLineThickness {
            get { return GetProperty(() => AverageLineThickness); }
            set { SetProperty(() => AverageLineThickness, value); }
        }
        public List<WeatherRecord> Data { get; private set; }
        public bool IsAverageWeather { get; private set; }
        public Color Color { get; private set; }
        public string Name { get; private set; }

        public WeatherItem(List<WeatherRecord> data, bool isAverageWeather, Color color, string name) {
            this.Data = data;
            this.IsAverageWeather = isAverageWeather;
            this.Color = color;
            this.Name = name;
            AverageLineThickness = 2;
        }
    }

    public class WeatherRecord {
        public static WeatherRecord Load(XElement element) {
            CultureInfo culture = CultureInfo.InvariantCulture;
            DateTime _date = DateTime.Parse(element.Attribute("Date").Value, culture);
            double min = Double.Parse(element.Attribute("Min").Value, culture);
            double max = Double.Parse(element.Attribute("Max").Value, culture);
            double avg = Double.Parse(element.Attribute("Avg").Value, culture);
            return new WeatherRecord(_date, max, avg, min);
        }

        DateTime date;

        public double MinValue { get; private set; }
        public double MaxValue { get; private set; }
        public double AvgValue { get; private set; }
        public DateTime Date { get { return date; } private set { date = value; } }

        WeatherRecord(DateTime _date, double maxValue, double avgValue, double minValue) {
            this.Date = _date;
            this.MaxValue = maxValue;
            this.AvgValue = avgValue;
            this.MinValue = minValue;
        }
    }
}

File: VB/ChartsDemo.Wpf/ViewModels/DailyWeatherViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Globalization
Imports System.Windows.Media
Imports System.Xml.Linq
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.Charts

Namespace ChartsDemo
    Public Class DailyWeatherViewModel
        Private Const vostokStationName As String = "Vostok Station"
        Private Const deathValleyName As String = "Death Valley, NV"


        Private ReadOnly weather_Renamed As List(Of WeatherItem)

        Public ReadOnly Property Weather() As List(Of WeatherItem)
            Get
                Return weather_Renamed
            End Get
        End Property

        Public Sub New()
            Dim valleyData As List(Of WeatherRecord) = LoadWeatherData("DeathValley.xml")
            Dim vostokData As List(Of WeatherRecord) = LoadWeatherData("VostokStation.xml")

            Dim palette As Palette = New OfficePalette()
            weather_Renamed = New List(Of WeatherItem)() From {
                New WeatherItem(valleyData, False, palette(1), deathValleyName),
                New WeatherItem(valleyData, True, palette(1), deathValleyName),
                New WeatherItem(vostokData, False, palette(0), vostokStationName),
                New WeatherItem(vostokData, True, palette(0), vostokStationName)
            }
        End Sub
        Private Function LoadWeatherData(ByVal fileName As String) As List(Of WeatherRecord)
            Dim items As New List(Of WeatherRecord)()
            Dim weatherDocument As XDocument = DataLoader.LoadXmlFromResources("/Data/DailyWeather/" & fileName)
            For Each element As XElement In weatherDocument.Root.Elements("Weather")
                items.Add(WeatherRecord.Load(element))
            Next element
            Return items
        End Function
        Public Overridable ReadOnly Property ChartAnimationService() As IChartAnimationService
            Get
                Return Nothing
            End Get
        End Property
        Public Sub OnModuleLoaded()
            If ChartAnimationService IsNot Nothing Then
                ChartAnimationService.Animate()
            End If
        End Sub
    End Class

    Public Class WeatherItem
        Inherits BindableBase

        Public Property AverageLineThickness() As Integer
            Get
                Return GetProperty(Function() AverageLineThickness)
            End Get
            Set(ByVal value As Integer)
                SetProperty(Function() AverageLineThickness, value)
            End Set
        End Property
        Private privateData As List(Of WeatherRecord)
        Public Property Data() As List(Of WeatherRecord)
            Get
                Return privateData
            End Get
            Private Set(ByVal value As List(Of WeatherRecord))
                privateData = value
            End Set
        End Property
        Private privateIsAverageWeather As Boolean
        Public Property IsAverageWeather() As Boolean
            Get
                Return privateIsAverageWeather
            End Get
            Private Set(ByVal value As Boolean)
                privateIsAverageWeather = value
            End Set
        End Property
        Private privateColor As Color
        Public Property Color() As Color
            Get
                Return privateColor
            End Get
            Private Set(ByVal value As Color)
                privateColor = value
            End Set
        End Property
        Private privateName As String
        Public Property Name() As String
            Get
                Return privateName
            End Get
            Private Set(ByVal value As String)
                privateName = value
            End Set
        End Property

        Public Sub New(ByVal data As List(Of WeatherRecord), ByVal isAverageWeather As Boolean, ByVal color As Color, ByVal name As String)
            Me.Data = data
            Me.IsAverageWeather = isAverageWeather
            Me.Color = color
            Me.Name = name
            AverageLineThickness = 2
        End Sub
    End Class

    Public Class WeatherRecord
        Public Shared Function Load(ByVal element As XElement) As WeatherRecord
            Dim culture As CultureInfo = CultureInfo.InvariantCulture
            Dim _date As Date = Date.Parse(element.Attribute("Date").Value, culture)
            Dim min As Double = Double.Parse(element.Attribute("Min").Value, culture)
            Dim max As Double = Double.Parse(element.Attribute("Max").Value, culture)
            Dim avg As Double = Double.Parse(element.Attribute("Avg").Value, culture)
            Return New WeatherRecord(_date, max, avg, min)
        End Function


        Private date_Renamed As Date

        Private privateMinValue As Double
        Public Property MinValue() As Double
            Get
                Return privateMinValue
            End Get
            Private Set(ByVal value As Double)
                privateMinValue = value
            End Set
        End Property
        Private privateMaxValue As Double
        Public Property MaxValue() As Double
            Get
                Return privateMaxValue
            End Get
            Private Set(ByVal value As Double)
                privateMaxValue = value
            End Set
        End Property
        Private privateAvgValue As Double
        Public Property AvgValue() As Double
            Get
                Return privateAvgValue
            End Get
            Private Set(ByVal value As Double)
                privateAvgValue = value
            End Set
        End Property
        Public Property [Date]() As Date
            Get
                Return date_Renamed
            End Get
            Private Set(ByVal value As Date)
                date_Renamed = value
            End Set
        End Property

        Private Sub New(ByVal _date As Date, ByVal maxValue As Double, ByVal avgValue As Double, ByVal minValue As Double)
            Me.Date = _date
            Me.MaxValue = maxValue
            Me.AvgValue = avgValue
            Me.MinValue = minValue
        End Sub
    End Class
End Namespace
